src/shell/interpreter.zig

```diff
@@ -2152,6 +2151,37 @@ pub const Interpreter = struct {
         }
 
         fn expandVar(this: *const Expansion, label: []const u8) EnvStr {
+            if (bun.strings.isAllASCIIDigit(label)) {
+                var int = std.fmt.parseInt(u32, label, 10) catch return EnvStr.initSlice("");
```

Comment:
It actually seems like bash only allows 1 digit positionals:

```bash
echo $123456
```

Prints: `23456`

Reply:
what platform are u on? I dont experience this

Reply:
<img width="178" alt="image" src="https://github.com/oven-sh/bun/assets/5464072/6167dad6-3bf7-4d39-b5e7-51a1b7349032">

Reply:
`sh` does reproduce what u said